website/source/docs/providers/azurerm/r/virtual_machine_scale_set.html.markdown

```diff
@@ -327,7 +325,7 @@ The following arguments are supported:
 
 `storage_profile_os_disk` supports the following:
 
-* `name` - (Required) Specifies the disk name.
+* `name` - (Required) Specifies the disk name. Value must be blank (`""`) when `managed_disk_type` is specified. 
```

Comment:
Related: I've opened issue #14932 to discuss making this field Optional, so this is no longer needed